### PR TITLE
[INLONG-7156][Agent] Support directly sending raw file data

### LINKED ISSUE #7156: [Improve][Agent] Support directly sending raw file data
### Description

_No response_

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pocozh**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- [INLONG-7156][Agent] Support directly sending raw file data

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #7156 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/task/TestTextFileTask.java [hunk 1]
```diff
@@ -180,7 +182,7 @@ public void testReadFull() throws IOException {
         jobProfile.set(JobConstants.JOB_DIR_FILTER_PATTERNS, file.getAbsolutePath());
         jobProfile.set(JobConstants.JOB_TASK_BEGIN_WAIT_SECONDS, String.valueOf(0));
         jobProfile.set(JobConstants.JOB_FILE_CONTENT_COLLECT_TYPE, DataCollectType.FULL);
-
+        jobProfile.set(JOB_FILE_META_ENV_LIST, KUBERNETES);
```
Comment (thesumery) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/task/TestTextFileTask.java:185]: Pls add an unit test for non metadata text read.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java [hunk 2]
```diff
@@ -278,6 +282,9 @@ public void resiterMeta(JobProfile jobConf) {
             return;
         }
         String[] env = jobConf.get(JOB_FILE_META_ENV_LIST).split(COMMA);
+        if (env.length > 0) {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:285]: Why is metadata required when there are more than one environment?

Comment (pocozh) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:285]: > Why is metadata required when there are more than one environment?

MetaData is used for some special scenes, such as k8s log.

Comment (healchow) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:285]: If the value of `JOB_FILE_META_ENV_LIST` is "local" or others, but not "k8s", then `needMetaData` will also be true. Is this as expected?

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java [hunk 3]
```diff
@@ -104,6 +104,7 @@ public class FileReaderOperator extends AbstractReader {
 
     private final BlockingQueue<String> queue = new LinkedBlockingQueue<>(CACHE_QUEUE_SIZE);
     private final StringBuffer sb = new StringBuffer();
+    private boolean needMetaData = false;
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:107]: `Metadata` is a complete word, so don't need to write it as `MetaData`.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java (hunk 2 above)
Comment (pocozh) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java]: > If the value of `JOB_FILE_META_ENV_LIST` is "local" or others, but not "k8s", then `needMetaData` will also be true. Is this as expected?

Okay, you are right

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**